R/plotreg.R

```diff
@@ -1,394 +1,372 @@
-      co.names <- dataframe[, 1]
-      co <- dataframe[, 2]
-      lower.inner <- dataframe[, 3]
-      upper.inner <- dataframe[, 4]
-      lower.outer <- dataframe[, 5]
-      upper.outer <- dataframe[, 6]
-      
-      signif.outer <- TRUE
-      note <- "Bars denote CIs."
-      message(paste0("Model ", i, ": bars denote 0.5 (inner) resp. ", ci.level, 
-          " (outer) confidence intervals (computed from standard errors)."))
+        z.inner <- qnorm(1 - ((1 - 0.5) / 2))
+        lower.inner <- dataframe$co - (z.inner * dataframe$se)
+        upper.inner <- dataframe$co + (z.inner * dataframe$se)
+        z.outer <- qnorm(1 - ((1 - ci.level) / 2))
+        lower.outer <- dataframe$co - (z.outer * dataframe$se)
+        upper.outer <- dataframe$co + (z.outer * dataframe$se)
+        
     } else if (length(models[[i]]@se) == 0 && length(models[[i]]@pvalues) > 0) {
-      stop("Model has p-values but no SEs. SEs or CIs are required for plotting.")
+        stop("Model has p-values but no SEs. SEs or CIs are required for plotting.")
     } else {
-      co <- models[[i]]@coef
-      co.names <- models[[i]]@coef.names
-      co <- co[-remove.rows]
-      co.names <- co.names[-remove.rows]
-      lower.outer <- models[[i]]@ci.low
-      upper.outer <- models[[i]]@ci.up
-      lower.inner <- NULL
-      upper.inner <- NULL
-      lower.outer <- lower.outer[-remove.rows]
-      upper.outer <- upper.outer[-remove.rows]
-      
-      # sort according to reorder.coef argument
-      dataframe <- data.frame(co.names, co, lower.outer, upper.outer)
-      dataframe <- reorder(dataframe, reorder.coef)
-      co.names <- dataframe[, 1]
-      co <- dataframe[, 2]
-      lower.outer <- dataframe[, 3]
-      upper.outer <- dataframe[, 4]
-      
-      signif.outer <- TRUE
-      note <- "Bars denote CIs."
-      message(paste0("Model ", i, ": bars denote  ", ci.level, 
-          " confidence intervals."))
+        lower.outer <- dataframe$ci.low
+        upper.outer <- dataframe$ci.up
+        lower.inner <- rep(0, length(dataframe$co.names))
+        upper.inner <- rep(0, length(dataframe$co.names))
+    }
+    
+    if (length(lower.outer) > 0) {
+        dataframe <- cbind(dataframe, lower.outer)     
+    }
+    if (length(lower.inner) > 0) {
+        dataframe <- cbind(dataframe, lower.inner)     
+    }
+    if (length(upper.outer) > 0) {
+        dataframe <- cbind(dataframe, upper.outer)     
+    }    
+    if (length(upper.inner) > 0) {
+        dataframe <- cbind(dataframe, upper.inner)     
     }
+    if (length(dataframe$pv) > 0) {
+        signif.outer <- dataframe$pv < (1 - ci.level)
+    } 
     
-    if (!is.null(custom.note)) {
-      note <- custom.note
+    # which terms are significant?
+    if(length(dataframe$ci.low) > 0) {
+        signif.outer <- ((dataframe$ci.low) > 0 & (dataframe$ci.up) > 0 | (dataframe$ci.low) < 0 & (dataframe$ci.up) < 0)
     }
     
+    if (class(signif.outer) == "logical" &&
```

Comment:
Is it logically possible that neither of the two conditions is true in which you create the `signif.outer` object? Just checking, because then this line would result in an uncaught error.

Reply:
since I took this bit from your code, I trusted it is correct. However, since I changed the way signif.outer is defined the question is still open. Is it possible to have a model with no pvalue and no CI? I can't think of such a model. Does it exist a case like that?

Reply:
Maybe some specifications of `lme4` since it is unclear which CI method is optimal. Not sure, but if we have a choice, let's prepare for this unlikely case.

Reply:
I added an error to address this